cmd/ig/daemon.go

```diff
@@ -100,11 +127,57 @@ func newDaemonCommand(runtime runtime.Runtime) *cobra.Command {
 			log.Warnf("reading config: %v", err)
 		}
 
+		tlsOptionsSet := 0
+		for _, tlsOption := range []string{serverKey, serverCert, clientCA} {
+			if len(tlsOption) != 0 {
+				tlsOptionsSet++
+			}
+		}
+
+		if tlsOptionsSet > 1 && tlsOptionsSet < 3 {
+			return fmt.Errorf(`
+missing at least one the TLS related options:
+	* tls-key: %q
+	* tls-cert: %q
+	* tls-client-ca: %q
+All these options should be set at the same time to enable TLS connection`,
+				serverKey, serverCert, clientCA)
+		}
+
+		var options []grpc.ServerOption
+
+		if serverKey != "" && serverCert != "" && clientCA != "" {
+			cert, err := tls.LoadX509KeyPair(serverCert, serverKey)
+			if err != nil {
+				return fmt.Errorf("loading TLS keypair: %w", err)
+			}
+
+			ca := x509.NewCertPool()
+			caBytes, err := os.ReadFile(clientCA)
+			if err != nil {
+				return fmt.Errorf("loading client CA certificate: %w", err)
+			}
+
+			if ok := ca.AppendCertsFromPEM(caBytes); !ok {
+				return errors.New("parsing client CA certificate")
+			}
+
+			tlsConfig := &tls.Config{
+				ClientAuth:   tls.RequireAndVerifyClientCert,
+				Certificates: []tls.Certificate{cert},
+				ClientCAs:    ca,
+			}
+
+			options = append(options, grpc.Creds(credentials.NewTLS(tlsConfig)))
+		} else if !strings.HasPrefix(socket, "unix") {
+			log.Warnf("no TLS configuration provided, communication between daemon and CLI will not be ciphered")
```

Comment:
```suggestion
			log.Warnf("no TLS configuration provided, communication between daemon and CLI will not be encrypted")
```
Easier to understand IMHO. Maybe even replace `daemon and CLI` with `client and server`.